tests/common/platform/device_utils_dpu.py

```diff
@@ -0,0 +1,123 @@
+"""
+Helper script for DPU  operations
+"""
+
+import time
+import pytest
+from tests.common.utilities import wait_until
+from tests.common.helpers.platform_api import chassis, module
+from tests.platform_tests.api.conftest import platform_api_conn
+from tests.common.devices.sonic import * #noqa
+from tests.common.helpers.assertions import pytest_assert
+
+@pytest.fixture(scope='function', autouse=True)
+def dpu_poweron_fixture(duthosts, enum_rand_one_per_hwsku_hostname, request, platform_api_conn):
+    """
+    Executes power on all DPUs
+    Returns:
+        Returns True or False based on all DPUs powered on or not
+    """
+
+    duthost = duthosts[enum_rand_one_per_hwsku_hostname]
+    num_modules = int(chassis.get_num_modules(platform_api_conn))
+    ip_address_list = []
+
+    for index in range(num_modules):
+        dpu = module.get_name(platform_api_conn, index)
+        ip_address_list.append(module.get_midplane_ip(platform_api_conn, index))
+        duthosts.shell("config chassis modules startup %s"%dpu)
+        time.sleep(2)
+
+    pytest_assert(wait_until(120, 30, 0, check_dpu_ping_status, duthost, ip_address_list),
+                          "Not all DPUs operationally up")
+
+def check_dpu_ping_status(duthost, ip_address_list):
+    """
+    Executes ping to all DPUs
+    Args:
+        duthost : Host handle
+        ip_address_list (list): List of all DPU ip addresses
+    Returns:
+        Returns True or False based on Ping is successfull or not to all DPUs
+    """
+
+    ping_count = 0
+    for ip_address in ip_address_list:
+        output_ping = duthost.command("ping -c 3 %s"%ip_address)["stdout_lines"]
+        for i in range(len(output_ping)):
+            if "0% packet loss" in output_ping[i]:
+                ping_count += 1
+
+    return (ping_count == len(ip_address_list))
```

Comment:
Looks like the function can return with False when any of the ping fails, no need to try the remaining ones.

Reply:
And the outer parentheses are not necessary.

Reply:
removed outer parentheses